Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportSessionCocoa.mm

```diff
@@ -37,121 +37,205 @@
-        RefPtr strongThis = weakThis.get();
-        if (!strongThis)
-            return;
+    m_connections.append(datagramConnection);
+    setupListenerConnectionHandler();
+    receiveDatagramLoop();
+}
 
-        // FIXME: Not only is this an unnecessary string copy, but it's also something that should probably be in WTF or FragmentedSharedBuffer.
-        auto vectorFromData = [](dispatch_data_t content) {
-            ASSERT(content);
-            Vector<uint8_t> request;
-            dispatch_data_apply_span(content, [&](std::span<const uint8_t> data) {
-                request.append(data);
-                return true;
-            });
-            return request;
-        };
+static RetainPtr<nw_parameters_t> createDatagramParameters()
+{
+    auto configureWebTransport = [](nw_protocol_options_t options) {
+        nw_webtransport_options_set_is_unidirectional(options, false);
+        nw_webtransport_options_set_is_datagram(options, true);
+    };
 
-        strongThis->receiveDatagram(vectorFromData(datagram).span());
-    }).get());
+    auto configureTLS = [](nw_protocol_options_t options) {
+        RetainPtr securityOptions = adoptNS(nw_tls_copy_sec_protocol_options(options));
+        sec_protocol_options_set_peer_authentication_required(securityOptions.get(), true);
+        sec_protocol_options_set_verify_block(securityOptions.get(), makeBlockPtr([](sec_protocol_metadata_t metadata, sec_trust_t trust, sec_protocol_verify_complete_t completion) {
+            // FIXME: Hook this up with WKNavigationDelegate.didReceiveChallenge.
+            completion(true);
+        }).get(), dispatch_get_main_queue());
+        // FIXME: Pipe client cert auth into this too, probably.
+    };
+
+    auto configureQUIC = [](nw_protocol_options_t options) {
+        nw_quic_set_initial_max_streams_bidirectional(options, 1000);
+        nw_quic_set_initial_max_streams_unidirectional(options, 1000);
+        nw_quic_set_max_datagram_frame_size(options, 65535);
+    };
 
-    // FIXME: Use nw_connection_group_set_new_connection_handler to receive incoming connections.
+    RetainPtr<nw_parameters_t> parameters = adoptNS(nw_parameters_create_webtransport_http(configureWebTransport, configureTLS, configureQUIC, NW_PARAMETERS_DEFAULT_CONFIGURATION));
+    return parameters;
 }
 
 void NetworkTransportSession::initialize(NetworkConnectionToWebProcess& connectionToWebProcess, URL&& url, CompletionHandler<void(RefPtr<NetworkTransportSession>&&)>&& completionHandler)
 {
-    // FIXME: Use nw_endpoint_create_url to support all URLs on systems where rdar://135036170 is fixed.
-    auto port = url.port() ? url.port() : defaultPortForProtocol(url.protocol());
-    if (!port)
-        return completionHandler(nullptr);
-
-    RetainPtr endpoint = adoptNS(nw_endpoint_create_host(url.host().utf8().data(), makeString(*port).utf8().data()));
+    RetainPtr<nw_endpoint_t> endpoint = adoptNS(nw_endpoint_create_url(url.string().utf8().data()));
```

Comment:
&lt;nw_endpoint_t&gt; isn't necessary because of the deduction guides of RetainPtr.  Ditto many times in this PR.